core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1049,6 +1052,7 @@ object KafkaConfig {
       .defineInternal(BrokerHeartbeatIntervalMsProp, INT, Defaults.BrokerHeartbeatIntervalMs, null, MEDIUM, BrokerHeartbeatIntervalMsDoc)
       .defineInternal(BrokerSessionTimeoutMsProp, INT, Defaults.BrokerSessionTimeoutMs, null, MEDIUM, BrokerSessionTimeoutMsDoc)
       .defineInternal(MetadataLogDirProp, STRING, null, null, HIGH, MetadataLogDirDoc)
+      .defineInternal(MetadataSnapshotMinNewRecordBytesProp, INT, Defaults.MetadataSnapshotMinNewRecordBytes, atLeast(1), HIGH, MetadataSnapshotMinNewRecordBytesDoc)
```

Comment:
This should be a public configuration.

Reply:
Good point. Can you write a follow-up PR to make this and other Raft configs public (for 3.0)?

Reply:
I'll make the property that I added public. The rest we can do as part of this Jira/PR: https://issues.apache.org/jira/browse/KAFKA-12992